lib/python/treadmill_aws/hostmanager.py

```diff
@@ -85,3 +85,92 @@ def find_hosts(ipa_client, pattern=None):
     return ipa_client.get_hosts(
         pattern=pattern
     )
+
+
+def is_space_available(subnets):
+    """ Returns total available IPs. """
+    total_available_ips = sum(
+        [subnet[0]['AvailableIpAddressCount'] for subnet in subnets]
+    )
+
+    return total_available_ips
+
+
+def get_availability(subnet):
+    """ Returns subnet`s total and available IPs. """
+    subnet_available_ips = subnet[0]['AvailableIpAddressCount']
+    subnet_cidr = subnet[0]['CidrBlock']
+    subnet_total_ips = pow(
+        2, (32 - int(subnet_cidr[subnet_cidr.find('/') + 1:]))
+    )
+
+    return subnet_available_ips, subnet_total_ips
+
+
+def get_availability_rate(placements):
+    """ Returns subnet`s availability rate. """
+    availability_rate = {}
+
+    for network, availability in placements.items():
+        availability_rate[network] = (availability[0] * 100) / \
+            availability[1]
+
+    availability_rate = [(v, k) for k, v in availability_rate.items()]
```

Comment:
this can be written as:

sorted(availability_rate.items(), key=lambda x: x[1])